qdev_wrappers/customised_instruments/heterodyne_source.py

```diff
@@ -0,0 +1,128 @@
+from qcodes.instrument.base import Instrument
+from qcodes.utils import validators as vals
+import logging
+from qdev_wrappers.parameters import DelegateParameter
+
+# TODO: name help pleaaase!
+# TODO: should the cavity and local os actually be parameters?
```

Comment:
Is is handy to be able to replace them? We probably want to make sure that it's captured in the snapshot which instruments they are

Reply:
bad thing is that lots of the other parameters of the instrument are delegate parameters and so if it is possible to change the values of the InstrumentRefParameter then you could end up in a situation where InstrumentRefParameter points to another instrument but the delegate parameters are still on the original. It seems like overengineering to do a workaround for that.